include/circt/Dialect/LLHD/IR/LLHDTypes.h

```diff
@@ -23,4 +23,14 @@
 #define GET_ATTRDEF_CLASSES
 #include "circt/Dialect/LLHD/IR/LLHDAttributes.h.inc"
 
+namespace llvm {
+template <>
+struct PointerLikeTypeTraits<circt::llhd::TimeAttr>
+    : public PointerLikeTypeTraits<mlir::Attribute> {
+  static inline circt::llhd::TimeAttr getFromVoidPointer(void *p) {
+    return circt::llhd::TimeAttr::getFromOpaquePointer(p);
+  }
+};
+} // namespace llvm
+
```

Comment:
Is that necessary for PointerUnion?

Reply:
Yes, it needs to know the number of available bits. It's unfortunate that TableGen doesn't produce this -- I've copied this snippet over from `mlir/IR/BuiltingAttributes.h` :disappointed: